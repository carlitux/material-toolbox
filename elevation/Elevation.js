'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Elevation = function Elevation(_ref) {
  var children = _ref.children,
      transition = _ref.transition,
      elevation = _ref.elevation;
  return React.Children.map(children, function (child) {
    var _classnames;

    var newClassName = (0, _classnames3.default)(child.props.className, (_classnames = {}, _defineProperty(_classnames, 'mdc-elevation--z' + elevation, elevation >= 0), _defineProperty(_classnames, 'mdc-elevation-transition', transition), _classnames));
    return React.createElement(child.type, _extends({}, child.props, {
      className: newClassName,
      ref: function ref(node) {
        var ref = child.ref;

        if (typeof ref === 'function') {
          ref(node);
        }
      }
    }));
  });
};

var _default = Elevation;
exports.default = _default;
;

var _temp = function () {
  if (typeof __REACT_HOT_LOADER__ === 'undefined') {
    return;
  }

  __REACT_HOT_LOADER__.register(Elevation, 'Elevation', 'src/elevation/Elevation.js');

  __REACT_HOT_LOADER__.register(_default, 'default', 'src/elevation/Elevation.js');
}();

;
module.exports = exports['default'];