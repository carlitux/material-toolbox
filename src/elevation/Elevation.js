// @flow
import * as React from 'react';
import classnames from 'classnames';

type Props = {
  children: React.Node,
  transition: boolean,
  elevation: number,
};

const Elevation = ({ children, transition, elevation }: Props) =>
  React.Children.map(children, child => {
    const newClassName = classnames(child.props.className, {
      [`mdc-elevation--z${elevation || ''}`]: elevation >= 0,
      'mdc-elevation-transition': transition,
    });
    return <child.type {...child.props} className={newClassName} />;
  });

export default Elevation;
