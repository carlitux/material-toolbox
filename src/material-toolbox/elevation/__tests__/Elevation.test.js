// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Elevation from '../index';

describe('component::Elevation', () => {
  for (let i = 1; i <= 24; i += 1) {
    it(`Should match snapshoot elevation-${i}`, () => {
      const component = shallow(
        <Elevation depth={i}>
          <div className="test" />
        </Elevation>,
      );
      expect(component).toMatchSnapshot();
    });
  }

  it(`Should match snapshoot transition with elevation 1`, () => {
    const component = shallow(
      <Elevation depth={1} transition>
        <div className="test" />
      </Elevation>,
    );
    expect(component).toMatchSnapshot();
  });
});
