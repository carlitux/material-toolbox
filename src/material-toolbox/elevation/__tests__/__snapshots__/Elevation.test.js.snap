// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::Elevation Should match snapshoot elevation-1 1`] = `
Array [
  <div
    className="test mdc-elevation--z1"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-2 1`] = `
Array [
  <div
    className="test mdc-elevation--z2"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-3 1`] = `
Array [
  <div
    className="test mdc-elevation--z3"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-4 1`] = `
Array [
  <div
    className="test mdc-elevation--z4"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-5 1`] = `
Array [
  <div
    className="test mdc-elevation--z5"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-6 1`] = `
Array [
  <div
    className="test mdc-elevation--z6"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-7 1`] = `
Array [
  <div
    className="test mdc-elevation--z7"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-8 1`] = `
Array [
  <div
    className="test mdc-elevation--z8"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-9 1`] = `
Array [
  <div
    className="test mdc-elevation--z9"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-10 1`] = `
Array [
  <div
    className="test mdc-elevation--z10"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-11 1`] = `
Array [
  <div
    className="test mdc-elevation--z11"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-12 1`] = `
Array [
  <div
    className="test mdc-elevation--z12"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-13 1`] = `
Array [
  <div
    className="test mdc-elevation--z13"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-14 1`] = `
Array [
  <div
    className="test mdc-elevation--z14"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-15 1`] = `
Array [
  <div
    className="test mdc-elevation--z15"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-16 1`] = `
Array [
  <div
    className="test mdc-elevation--z16"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-17 1`] = `
Array [
  <div
    className="test mdc-elevation--z17"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-18 1`] = `
Array [
  <div
    className="test mdc-elevation--z18"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-19 1`] = `
Array [
  <div
    className="test mdc-elevation--z19"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-20 1`] = `
Array [
  <div
    className="test mdc-elevation--z20"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-21 1`] = `
Array [
  <div
    className="test mdc-elevation--z21"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-22 1`] = `
Array [
  <div
    className="test mdc-elevation--z22"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-23 1`] = `
Array [
  <div
    className="test mdc-elevation--z23"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot elevation-24 1`] = `
Array [
  <div
    className="test mdc-elevation--z24"
    key=".0"
  />,
]
`;

exports[`component::Elevation Should match snapshoot transition with elevation 1 1`] = `
Array [
  <div
    className="test mdc-elevation--z1 mdc-elevation-transition"
    key=".0"
  />,
]
`;
